### DISCUSSION TITLE
Order of search results on dedicated search page

### DISCUSSION THREAD

**Discussion by mnissl** [McShelby/hugo-theme-relearn#483] [Q&A] [answered]
Is it possible to set the sort order of search results on the dedicated search page?

Right now, results are presented in an arbitrary order, at least it looks like this to me.

IMHO, respecting the order as given by the menu entries would make sense.

**Comment by McShelby** [comment]
Although there are plans to add more options to the dedicated search page (#407), ordering is currently solely done by Lunr based on the configuration we are setting.

Lunr calculates a weight based on these settings and sorts accordingly.

This will not change even more options are implemented. So the only way to achieve this is by setting diddferent configuration settings for Lunr.

Nevertheless I doubt, that sorting by menu order is a good idea for search results as the pages with a higher weight may get listed far at the bottom.

**Reply by mnissl** [reply]
> Nevertheless I doubt, that sorting by menu order is a good idea for search results as the pages with a higher weight may get listed far at the bottom.

I cannot follow this reasoning: if something is further down the menu, wouldn't that be okay as well to be further down the search results page?

Of course the sort order should be configurable as one solution certainly won't fit all.

My use case of Hugo with your theme is a collection of release notes: the top level menu entries show the major version number of the application from the latest version on top to the oldest version at the very bottom. Being able to search through all release notes is a great benefit ... alas, the search result page shows the result in "abritrary order". My customers (including me) would expect them from latest (on top) to oldest (at the bottom).

Sorting by article date would be an option as well. I didn't bother so far filling in a publish date in the front matter as I have one page for each minor release which features all bug fix releases ... so actually, I have several publish dates. But going for the earliest or latest release date of a bug fix release would do the job for the publish date in the front matter.

**Reply by McShelby** [reply]
> I cannot follow this reasoning: if something is further down the menu, wouldn't that be okay as well to be further down the search results page?

Not necessarily. Consider you have released a new feature in the past and have initial release notes describing the feature in detail. Further release notes may only briefly mention a bugfix. If you are just looking for the feature description you want that to show up and not the mentioning of bugfixes. With a weighted approach, this is most likely to happen, because the search term may be used multiple times in the initial release notes resulting in a higher weight.

I agree that sorting by creation / change date is a desireable option.

**Reply by mnissl** [reply]
By "weight" you mean the number of hits/matches, not the weight option in the front matter 😉 

> I agree that sorting by creation / change date is a desireable option.

Great. Is this something I could hardcode in my copy of your theme until you make it configurable in an upcoming release? Or do you intend to implement this option in a release soon? So, if not, what am I supposed to change?

**Reply by McShelby** [reply, marked as answer]
"Soon" as in "soon™" ;-)

Seriously, I do not plan to implement it in the next three month. But you never know - sometimes priorities shift. Nevertheless, this is very high on my list - after I fixed a few other things. So realisticly this will come somewhere around this year.

If you want to take a look into this, `static/search.js` is the place of interest. That's where Lunr is initialized. You'll most likly need to take a look into their documentation. If you need furhter data for the search index, you need to change `layouts/_default/index.json` (deprecated but still supported) and `layouts/_default/index.search.js` (the hot new stuff).

**Reply by mnissl** [reply]
It seems that it is a little bit more complicated than simply altering one command line call. So if you implement this feature "somewhere around this year", then this will be "soon" enough 😄 Thanks.
